CONTRIBUTING.md

```diff
@@ -90,6 +90,16 @@ We recommend the following commit structure in the following order:
 
 Follow [this guide](https://chris.beams.io/posts/git-commit/) to ensure you're writing good commit messages.
 
+## Creating a Release
+
+To create a production release of mbed-tools, perform the following steps:
+
+1. Check out the `master` branch and ensure it is up to date.
+1. Run `tox -e preprelease`, this will update the necessary files, create a commit and tag it with the new release version number.
+1. Push the tag to the repository.
```

Comment:
We also need to push to the master branch, to get the "Prepare mbed-tools release" commit published. Just pushing the tag isn't enough to also update the master branch.

Reply:
We should push to master first. Otherwise, if we push to master after pushing the tag, we get the following error from PyPI as the release for this version has already been completed (when the tag was pushed).

```
adding 'mbed_tools-3.5.0.dist-info/WHEEL'
adding 'mbed_tools-3.5.0.dist-info/entry_points.txt'
adding 'mbed_tools-3.5.0.dist-info/top_level.txt'
adding 'mbed_tools-3.5.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
pypirelease run-test: commands[1] | python -m twine check 'dist/*'
Checking dist/mbed_tools-3.5.0-py3-none-any.whl: PASSED
Checking dist/mbed-tools-3.5.0.tar.gz: PASSED
pypirelease run-test: commands[2] | python -m twine upload -p *** -u *** 'dist/*'
Uploading distributions to https://upload.pypi.org/legacy/
Uploading mbed_tools-3.5.0-py3-none-any.whl

  0%|          | 0.00/125k [00:00<?, ?B/s]
  6%|▋         | 8.00k/125k [00:00<00:04, 29.1kB/s]
100%|██████████| 125k/125k [00:00<00:00, 131kB/s]  HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
File already exists. See https://pypi.org/help/#file-name-reuse for more information.

NOTE: Try --verbose to see response content.
ERROR: InvocationError for command /home/vsts/work/1/s/.tox/pypirelease/bin/python -m twine upload -p *** -u *** 'dist/*' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   pypirelease: commands failed
##[error]Bash exited with code '1'.
Finishing: PyPI release

```